### PR TITLE
Remove lazy loaded members when leaving room

### PR DESCRIPTION
**Pull request by bwindels**
Fixes https://github.com/vector-im/riot-web/issues/7226, but needs more testing so don't merge yet.

### CODE REVIEW

**Comment by bwindels** [comment]
Did more testing, seems to work well.

File: src/store/indexeddb-local-backend.js [hunk 1]
```diff
@@ -292,6 +303,44 @@ LocalIndexedDBStoreBackend.prototype = {
...
+            [minStateKeyProm, maxStateKeyProm]);
+
+        const writeTx = this.db.transaction(
+            ["oob_membership_events"],
+            "readwrite");
+        const writeStore = writeTx.objectStore("oob_membership_events");
+        const membersKeyRange = IDBKeyRange.bound(
+            [roomId, minStateKey],
+            [roomId, maxStateKey],
+        );
```
Comment (dbkr) [inline src/store/indexeddb-local-backend.js]: I'm missing why you can't just delete with `IDBKeyRange.only(roomId)`

Comment (bwindels) [reply src/store/indexeddb-local-backend.js]: The primary key of `oob_membership_events` is a composed key of `[roomId, stateKey]`, as you need those two to uniquely identify a member. I didn't try this, but I don't think `IDBKeyRange.only` supports matching only a part of a composed key.

Comment (bwindels) [reply src/store/indexeddb-local-backend.js]: Actually, I'm quite certain `IDBKeyRange.only` doesn't support prefix-matches as they are called, on composed keys; I asked this question earlier on stackoverflow to make sure this approach would work across browsers, and the person awnsering said prefix-matches are on peoples wish-lists for idb.

https://stackoverflow.com/questions/52111115/safe-to-assume-same-order-in-index-as-store-with-upper-part-of-key-same/52120855#52120855

**Pull request merged**